web/html/src/manager/virtualization/nets/network-properties.js

```diff
@@ -0,0 +1,432 @@
+                    invalidHint={t('The value has to be a positive integer')}
+                  />)
+                }
+                { FieldsData.getValue(model.type, 'virtualport_types', undefined) && (
+                  <>
+                    <Select
+                      labelClass="col-md-3"
+                      divClass="col-md-6"
+                      label={t('Virtual Port Type')}
+                      name="virtualport_type"
+                      key={`virtualport_type_${model.type}`}
+                      options={
+                        FieldsData.getValue(model.type, 'virtualport_types', []).map(label => (
+                          {
+                            value: label.toLowerCase().replaceAll(' ', ''),
+                            label: label,
+                          }
+                        ))
+                      }
+                      isClearable
+                    />
+                    <VirtualPortFields type={model.virtualport_type}/>
+                  </>)
+                }
+                { FieldsData.getValue(model.type, 'pf', false) && (
+                    <Radio
+                      name="interface-selection-type"
+                      inline={true}
+                      label={t('Select interfaces')}
+                      required
+                      labelClass="col-md-3"
+                      divClass="col-md-6"
+                      defaultValue='vf'
+                      items={[
+                        {label: t('By physical function'), value: 'pf'},
+                        {label: FieldsData.getValue(model.type, 'vf', false) ? t('By virtual functions') : t('By interfaces'), value: 'vf'},
+                      ]}
+                    />
+                  )
+                }
+                { model['interface-selection-type'] !== 'pf' && FieldsData.getValue(model.type, 'interfaces', false) && (
+                  <>
+                    <Select
+                      labelClass="col-md-3"
+                      divClass="col-md-6"
+                      label={t('Interfaces')}
+                      name="interfaces"
+                      options={netDevices}
+                      getOptionValue={option => option != null ? option.name : ""}
+                      formatOptionLabel={(option, {context}) => {
+                        return <Interface device={option} short={context === "value"}/>;
+                      }}
+                      isMulti
+                    />
+                    {
+                      (model.interfaces || []).map(nic => (
+                        <div className="col-md-6 col-md-offset-3">
+                          <Interface device={netDevices.find(dev => dev.name === nic)} />
+                        </div>
+                      ))
+                    }
+                  </>
+                )}
+                { model['interface-selection-type'] === 'pf' && FieldsData.getValue(model.type, 'pf', false) && (
```

Comment:
I would recommend extracting these bigger blocks out of the template into a boolean variable with a descriptive name a-la `showPhysicalFunction`. Right now it's a bit of a mouthful to read the template since there's a lot of logic and view stuff mixed in in long goes.

Reply:
ACK